services-directions-refresh-models/src/main/java/com/mapbox/api/directionsrefresh/v1/models/DirectionsRefreshJsonObject.java

```diff
@@ -25,4 +34,51 @@ public String toJson() {
     gson.registerTypeAdapterFactory(DirectionsRefreshAdapterFactory.create());
     return gson.create().toJson(this);
   }
+
+  /**
+   * Use this method to get names of JSON properties that weren't recognized during JSON
+   * serialization by the model. This may be useful to access experimental API properties
+   * via {@link #getUnrecognizedProperty(String)}.
+   * When an experimental API property becomes stable,
+   * it will eventually have static field in a model introduced
+   * and it won't be available via this dynamic method anymore.
+   *
+   * See
+   * <a href="https://docs.mapbox.com/api/navigation/directions/">Directions API documentation</a>
+   * for available experimental fields.
+   *
+   * @return names of unrecognized JSON properties or an empty set
+   */
+  @NonNull
+  public final Set<String> getUnrecognizedPropertiesNames() {
+    return UnrecognizedPropertiesUtils.getUnrecognizedPropertiesNames(unrecognized());
+  }
+
+  /**
+   * Use this method to access a JSON property that wasn't recognized during JSON serialization
+   * by the model. This may be useful to access experimental API properties.
+   * When an experimental API property becomes stable,
+   * it will eventually have static field in a model introduced
+   * and it won't be available via this dynamic method anymore.
+   *
+   * See
+   * <a href="https://docs.mapbox.com/api/navigation/directions/">Directions API documentation</a>
+   * for available experimental fields.
+   *
+   * @param propertyName name of a json property
+   * @return value of a requested property or null if the requested property doesn't exist.
+   */
+  @Nullable
+  public final JsonElement getUnrecognizedProperty(String propertyName) {
+    return UnrecognizedPropertiesUtils.getUnrecognizedProperty(propertyName, unrecognized());
+  }
+
+  @Nullable
+  @UnrecognizedJsonProperties
+  abstract Map<String, SerializableJsonElement> unrecognized();
+
+  abstract static class Builder<T extends Builder> {
+    @NonNull
+    abstract T unrecognized(@Nullable Map<String, SerializableJsonElement> value);
```

Comment:
```suggestion
    internal abstract T unrecognized(@Nullable Map<String, SerializableJsonElement> value);
```
Let's keep it hidden from the end user

Reply:
Absent modifier means package-private here. I did it like this to keep it consistent with `DirectionsJsonObject`.
But the question is why it should be hidden at all.